labs/lab06.md

```diff
@@ -0,0 +1,75 @@
+# Lab 6: Refactoring and Anti-Patterns
+
+## Introduction 
+A lot of the design principles and heuristics we have been encouraging in this class are there to encourage good design *up-front*. But you do not always have control over how a system you are using gets created. 
+
+
+**Anti-patterns** are common examples of bad design. They tend to come about due to a program’s natural evolution, coupled with poor choices early on, which leads to code that is increasingly *tangled*, badly coupled, low on cohesion, etc. These are often signaled by **code smells** -- heuristic, but not guaranteed indicators of anti-patterns, like overly long methods, or a class that calls a lot of another class’ code (feature envy, signals high coupling).
+
+In this lab, you will become a code-smelling agent who detects  and attempt at refactoring them. 
+## Deliverables
```

Comment:
if you want to keep it fine, but I dislike the term code smell and would at most use it once above in a definition